### PR TITLE
Add doc/buip-hf.md guidance document

### PR DESCRIPTION
**Pull request by ftrader**
Add a document to provide guidance to those working on this hard fork development.

### CODE REVIEW

**Comment by ftrader** [comment]
I cancelled the Travis checks as this is a doc-only PR.

**Review by sickpig — APPROVED** [review]

File: doc/buip-hf.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+BUIP-HF Development Notes
+===========================
+
+Specification
+----------------
+
+There is a draft specification currently located at
+
+https://github.com/BitcoinUnlimited/BUIP/blob/master/BUIP-UAHF/buip-uahf-technical-spec.md
```
Comment (ftrader) [inline doc/buip-hf.md]: This is outdated I think. I will review / revise if needed.

Comment (sickpig) [reply doc/buip-hf.md]: you are right, cause we renamed the dir, this is the correct one:

https://github.com/BitcoinUnlimited/BUIP/blob/master/BUIP-HF/buip-hf-technical-spec.md

Comment (ftrader) [reply doc/buip-hf.md]: Thanks, I will update the PR shortly.

**Pull request merged**